lib/api/api.go

```diff
@@ -499,6 +519,16 @@ func (s *service) CommitConfiguration(from, to config.Configuration) bool {
 	return true
 }
 
+func (s *service) IsAuthEnabled() bool {
+	// This function should match isAuthEnabled() in syncthingController.js
+	guiCfg := s.cfg.GUI()
```

Comment:
Similar to the constructor comment on config, this method should take a gui configuration as a parameter to represent the current state of config as used by the api service/avoid config racyness.

Reply:
Thanks, should be fixed by https://github.com/emlun/syncthing/pull/4